### ISSUE TITLE
Enable bech32 receive address

### ISSUE THREAD

**Issue by mikeroberts76** [ACINQ/eclair-mobile#171]
Please, with bech32 addresses fees are lower.

**Comment by mikeroberts76** [comment]
Selection of "native" & "non-native" receive address for on-chain funding would be very nice

**Comment by sstone** [comment]
Yes we would like to support bech32 addresses (we chose p2sh-of-p2wpkh because there are still many wallets and services that don't understand bech32), but it's not scheduled yet and not that simple (how do we offer this option and migrate existing wallets, deal with pending closing txs, ....) so we cannot announce a release date.

**Comment by Iffabled** [comment]
@dpad85 sorry for mentioning you, but is there any ETA for this?

**Comment by maflcko** [comment]
As mentioned in the previous comment, this needs a major rewrite of the wallet, I think.

**Comment by cyanlink** [comment]
@MarcoFalke I really don't think so... ACINQ already released their brand new 2nd-gen wallet called Phoenix, which will only generate Bech32 addresses. They already got corresponding libraries too. The famous wasabi wallet featuring CoinJoin only scans/recognizes Bech32 Addresses. Even BLW (Bitcoin Lightning Wallet, also present on bitcoin.org's wallet list, with only 1 major contributor) generates Bech32 addresses. Eclair mobile is (a little bit) left behind...

**Comment by dpad85** [comment]
As mentioned by @sstone in the comment above, the issue is how to migrate existing wallets to bech32. Ideally we'd have an automated migration system that would take care of that behind the scene, but that's not easy to do. 

I think a fair solution would be that, from now on, every newly created wallet would be a bech32 wallet. Users willing to migrate would have to do that manually. We would have to add some instructions in the app to describe the process, which is:

1. Make sure all channels are closed
2. Send all funds somewhere else
3. Reset data/uninstall
4. Create new wallet

Also the "Restore wallet" page would need to know what kind of seed the user is trying to restore, which means adding a selection box in the UI with some very technical content that will confuse newcomers. But it's worth it.

**Comment by Kylejustknows** [comment]
why don't we treat the old p2wpk like an "old address". 
The seed generates 2 private keys, one for p2wpk, one for bech32, in the background. 

If the user is an old user, has fund/channel in old p2wpk, then run 2 processes, transparently dealing both, and always gives the user the bech32 receiving address as default, to promote user slowly move to bech32.

If the user is new, has no money no channel in the p2wpk private key, great! Simply run the bech32 process only from now on.
